simulator/memory/argv_loader/argv_loader.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * argv_loader.h: loader of arguments to guest memory
+ * @author Rustem Yunusov <yunusov.rr@phystech.edu>
+ * Copyright 2019 MIPT-MIPS iLab project
+ */
+
+#ifndef ARGV_LOADER_H
+#define ARGV_LOADER_H
+
+#include <infra/types.h>
+#include <memory/memory.h>
+#include <infra/argv.h>
+#include <infra/byte.h>
+
+struct ArgvLoaderError : Exception
+{
+    explicit ArgvLoaderError(const std::string& msg) :
+            Exception("Error while loading arguments to guest memory", msg) { }
+};
+
+struct InvalidArgs : std::invalid_argument
+{
+    explicit InvalidArgs(const std::string& msg) :
+            invalid_argument( msg) { }
+};
+
+class ArgvLoader
+{
+public:
+    explicit ArgvLoader( const char* const* argv, const char* const* envp = nullptr);
+    ~ArgvLoader();
+
+    // We delete everything explicitly
+    ArgvLoader( const ArgvLoader&) = delete;
+    ArgvLoader( ArgvLoader&&) = delete;
+    ArgvLoader& operator=( const ArgvLoader&) = delete;
+    ArgvLoader& operator=( ArgvLoader&&) = delete;
```

Comment:
I would use Rule of Zero: https://en.cppreference.com/w/cpp/language/rule_of_three.
All of the 5 special functions are trivial, so we can rely on compiler.

Reply:
So I should mark them 'default' instead of 'delete', right?

Reply:
No, just to do not declare them at all